### PR TITLE
DApp Bug Fixes

### PR DESCRIPTION
**Pull request by shahthepro**
[Link to test build](https://originprotocol.github.io/test-builds/shah/#/)

Should fix and close the following issues:
- #3433
- #3620 
- #3761 
- #3928 

### Checklist:

- [ ] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: dapps/marketplace/src/pages/transactions/_Purchase.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      )}
+    </div>
+    <div className="details">
+      <div className="top">
+        <h2 className="title mb-1">
+          {listing.title || <i>Untitled Listing</i>}
+        </h2>
+        <div className="right">
+          <span className="time-estimate">
+            {distanceToNow(offer.createdEvent.timestamp, true) + ' ago'}
```
Comment (evgenyboxer) [inline dapps/marketplace/src/pages/transactions/_Purchase.js]: shouldn't the string `ago` also use `fbt`?

File: dapps/marketplace/src/components/BottomScrollListener.js [hunk 2]
```diff
@@ -65,19 +65,40 @@ const BottomScrollListener = ({
...
+      canFetchMore = true
+    }
+
+    if (canFetchMore) {
       onBottom()
     }
   })
 
+  useEffect(() => {
+    const debounced = debounce(onWindowResize, 300)
```
Comment (evgenyboxer) [inline dapps/marketplace/src/components/BottomScrollListener.js]: minor - would be great to have the `300ms` as a constant.

Comment (shahthepro) [reply dapps/marketplace/src/components/BottomScrollListener.js]: Can be done. I don't think we have a constants file for DApp. So, Not prioritising this now. Will comeback to it later.

**Comment by shahthepro** [comment]
@nick @sparrowDom Merging this since I added a changeset that should fix #3984. Let me know if you any comments on anything. Will address them later.

**Pull request merged**